docs/index-type-operator.md

```diff
@@ -0,0 +1,88 @@
+# `index` type operators
+
+## Summary
+
+This RFC proposes the addition of one type operator, `index`, which can be used to look up a specific property of another type (like TypeScript's Indexed Access Type).
+
+## Motivation
+
+The primary motivation of this proposal is to allow Luau to define a type by accessing a property of another existing type. For instance, consider the following example code:
+```lua
+type Person = {
+  age: number,
+  name: string,
+  alive: boolean
+}
+
+local bob: Person = {
+  age = 22,
+  name = "Bob",
+  alive = true
+}
+
+local function doSmt(param: typeof(bob["age"])) -- param = string
+  -- rest of code
+end
+
+type unionType = typeof(bob["age"]) | typeof(bob["name"]) | typeof(bob["alive"]) -- unionType = number | string | boolean
+```
+
+This is a valid Luau program; however, in order to define the type of `Person["age"]` we had to first declare a variable `bob` and utilize the `typeof` type operator. This is quite cumbersome when developers want to typecheck using the type of `Person["age"]` without having to declare a variable first. Additionally, in order to define the union type of all the properties of `Person`, current Luau requires an explicit list of each property using `typeof`.
+
+The expected outcome of the index type operator is that it will enhance the developing experience and allow the Roblox community to develop better type-checked programs.
```

Comment:
Luau is not specific to Roblox, so we'd prefer not to bring it up unnecessarily.
```suggestion
The expected outcome of the index type operator is that it will enhance developer experience and allow Luau developers to more easily develop well-typed programs.
```